packages/js/product-editor/src/components/iframe-editor/header-toolbar/header-toolbar.tsx

```diff
@@ -87,42 +109,93 @@ export function HeaderToolbar( {
 		}
 	}, [ isInserterOpened, setIsInserterOpened ] );
 
+	useEffect( () => {
+		// If we have a new block selection, show the block tools
+		if ( hasBlockSelection ) {
+			setIsBlockToolsCollapsed( false );
+		}
+	}, [ hasBlockSelection ] );
+
+	const renderBlockToolbar =
+		isWpVersion( '6.5', '>=' ) || getGutenbergVersion() > 17.3;
```

Comment:
Checking the versions was unfortunately the only way to determine the new header approach.